tests/console/journalctl.pm

```diff
@@ -7,21 +7,115 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Test basic journalctl functionality (assert bsc#1063066 is not present)
-# - Verify "man -P cat journalctl" for broken man page format
-# Maintainer: Sergio Lindo Mansilla <slindomansilla@suse.com>
-# Tags: bsc#1063066
+# Summary: Test basic journalctl functionality
+# - assert bsc#1063066 is not present (Verify "man -P cat journalctl" for broken man page format)
+# - setup persistent journal
+# - setup FSS, rotate log, verify, reboot and verify again (See bsc#1171858)
+# - check if log entry from previous boots are present
+# - check if log filtering by time works
+# - check if journalctl kernel messages are present (non-empty output)
+# - check if redirect to serial port is working
+# - check if redirect to syslog is working (if applicable)
+# - check if journalctl vacuum functions are working
+# - verify FSS log again
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+#             Sergio Lindo Mansilla <slindomansilla@suse.com>
+# Tags: bsc#1063066 bsc#1171858
 
 use base "consoletest";
 use strict;
 use warnings;
 use testapi;
+use utils;
+use version_utils;
+use power_action_utils 'power_action';
+
+sub check_journal {
+    my ($args, $filename, $fail_message) = @_;
+    $fail_message //= "journalctl '$args' is empty";
+    assert_script_run("journalctl -q $args > $filename");
+    assert_script_run("if [ -s $filename ]; then true; else false; fi", fail_message => "$fail_message");
+}
+
+sub check_syslog {
+    # rsyslog is not installed on tumbleweed anymore
+    return !is_tumbleweed;
+}
 
 sub run {
+    my $self = shift;
```

Comment:
Sorry, I missed this one.

```suggestion
    my ($self) = @_;
```

Reply:
Included and updated the test runs accordingly